src/credential/Credential.ts

```diff
@@ -0,0 +1,72 @@
+import * as gabi from '@kiltprotocol/portablegabi'
+import IRequestForAttestation from '../types/RequestForAttestation'
+import IAttestation from '../types/Attestation'
+import Identity from '../identity/Identity'
+import AttestedClaim from '../attestedclaim/AttestedClaim'
+import ICredential from '../types/Credential'
+
+export default class Credential {
+  public readonly reqForAtt: IRequestForAttestation
+  public readonly attestation: IAttestation
+  public readonly privacyCredential: gabi.Credential | null
+
+  public static async fromRequestAndAttestation(
+    claimer: Identity,
+    request: IRequestForAttestation,
+    attestation: IAttestation,
+    session: gabi.ClaimerAttestationSession | null = null,
+    attestationPE: gabi.Attestation | null = null
+  ): Promise<Credential> {
+    let privacyCredential: gabi.Credential | null = null
+
+    if (session !== null && attestationPE !== null) {
+      privacyCredential = await claimer.claimer.buildCredential({
+        claimerSession: session,
+        attestation: attestationPE,
+      })
+    }
+    return new Credential({
+      reqForAtt: request,
+      attestation,
+      privacyCredential,
+    })
+  }
+
+  protected constructor({
+    reqForAtt,
+    attestation,
+    privacyCredential,
+  }: ICredential) {
+    this.reqForAtt = reqForAtt
+    this.attestation = attestation
+    this.privacyCredential = privacyCredential
+  }
+
+  public createPresentation(
+    excludedClaimProperties: string[],
+    excludeIdentity = false
+  ): AttestedClaim {
+    const attClaim = new AttestedClaim(
+      // clone the attestation and request for attestation because properties will be deleted later.
+      // TODO: find a nice way to clone stuff
```

Comment:
Not sure what's the best way to deeply clone an object. this was the best I found. Might be cool to have something different?

Reply:
Unfortunately, this method is not safe until we have type checking, see below. I would propose to add custom code or Really Fast Deep Clone [rdfc](https://www.npmjs.com/package/rfdc) which is like 400% faster than `lodash.deepClone`

```javascript
  const orig = {
    date: new Date()
  }
  const copy = JSON.parse(JSON.stringify(orig))
  // typeof orig.date === 'object'
  // typeof copy.date === 'string'
```

Reply:
In case of dates, we would rather use a Unix timestamp. So I think it is fine to use json parse & stringify, since it is also not performance critical.